src/config_generator/from_proto.rs

```diff
@@ -0,0 +1,356 @@
+    let package = helper.package.clone();
+    let mut grpc_method = GrpcMethod { package, service: "".to_string(), name: "".to_string() };
+    let mut ty = config.types.get(key).cloned().unwrap_or_default();
+
+    for service in services {
+        let service_name = service.name().to_string();
+        for method in &service.method {
+            let method_name = method.name();
+
+            helper.insert(method_name, DescriptorType::Query);
+
+            let mut cfg_field = Field::default();
+            if let Some((k, v)) = get_arg(method.input_type(), helper) {
+                cfg_field.args.insert(k, v);
+            }
+
+            let (output_ty, required) = get_output_ty(method.output_type());
+            cfg_field.type_of = helper.get(&output_ty).unwrap_or(output_ty.clone());
+            cfg_field.required = required;
+
+            grpc_method.service = service_name.clone();
+            grpc_method.name = method_name.to_string();
+
+            cfg_field.grpc = Some(Grpc {
+                base_url: None,
+                body: None,
+                group_by: vec![],
+                headers: vec![],
+                method: grpc_method.to_string(),
+            });
+            ty.fields
+                .insert(helper.get(method_name).unwrap(), cfg_field);
+        }
+    }
+    Ok(ty)
+}
+
+fn append_query_service(
+    config: &mut Config,
+    services: Vec<ServiceDescriptorProto>,
+    query: &str,
+    helper: &mut Helper,
+) -> anyhow::Result<()> {
+    if services.is_empty() {
+        return Ok(());
+    }
+
+    let ty = generate_ty(config, services, helper, query)?;
+
+    if ty.ne(&Type::default()) {
+        config.schema.query = Some(query.to_owned());
+        config.types.insert(query.to_owned(), ty);
+    }
+    Ok(())
+}
+
+pub fn build_config(
+    descriptor_sets: Vec<FileDescriptorSet>,
+    query: Option<String>,
+) -> anyhow::Result<Config> {
+    let mut config = Config::default();
+    let mut helper = Helper::default();
+    let query = query.unwrap_or_else(|| "Query".to_string());
+
+    for descriptor_set in descriptor_sets {
+        for file_descriptor in descriptor_set.file {
+            helper.package = file_descriptor.package().to_string();
+
+            append_enums(&mut config, file_descriptor.enum_type, &mut helper)?;
+            append_msg_type(&mut config, file_descriptor.message_type, &mut helper)?;
+            append_query_service(
+                &mut config,
+                file_descriptor.service.clone(),
+                &query,
+                &mut helper,
+            )?;
+        }
+    }
+
+    Ok(config)
+}
+
+#[cfg(test)]
+mod test {
+
+    use std::path::PathBuf;
+
+    use prost_reflect::prost_types::{FileDescriptorProto, FileDescriptorSet};
+
+    use crate::config_generator::from_proto::build_config;
+
+    fn get_proto_file_descriptor(name: &str) -> anyhow::Result<FileDescriptorProto> {
+        let mut proto_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
```

Comment:
Keep the test protofiles for the config generator in the config_generator folder